source-firestore/pull.go

```diff
@@ -577,6 +580,10 @@ func (c *capture) StreamChanges(ctx context.Context, client *firestore_v1.Client
 					return fmt.Errorf("unexpected target ID %d", tc.TargetIds[0])
 				}
 			case firestore_pb.TargetChange_REMOVE:
+				if catchupStreaming && time.Since(catchupStarted) > 5*time.Minute {
```

Comment:
It's not immediately obvious to me why this is 5 minutes, and why it is happening at `TargetChange_REMOVE`. I could surmise that `TargetChange_REMOVE` is what happens when we get cut off for taking too long. Any reason for this being 5 minutes instead of 10 minutes?

Reply:
This is pretty much just a heuristic that seemed reasonable to me. What I'm trying to do is detect the specific scenario of "there is a ton of data and we failed to catch up in time", but there are potentially other reasons Firestore might do a `TargetChange.REMOVE` to us, so I put in a heuristic of "it's been a little while" around that.

The precise value is kind of unimportant, but I don't want it to be too close to zero or to the actual threshold, because either of those seemed more likely to result in false positives or negatives. So I went with the midpoint.

Reply:
Ah ok - makes sense! Thanks.